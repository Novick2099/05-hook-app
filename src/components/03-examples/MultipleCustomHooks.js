import {useFetch} from "../../hooks/useFetch";
import {useCounter} from "../../hooks/useCounter";
import {Button} from "@material-ui/core";

const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1);
    const {loading, data} = useFetch(`https://breakingbadapi.com/api/quotes/${counter}`);
    //todo !niega el null  a true y si se vuelve a negar regresara false
    const {author, quote} = !!data && data[0];
    

    return (
        <>
            <div>
                <h1>Breaking bdas Quotes</h1>
                <p>Counter{counter}</p>
            </div>
            { loading
                ?
                (<div data-id={"alert"}>
                    Loading..
                </div>) :
                (<div>
                        <p data-id={"author"}>{author}</p>
                        <p data-id={"quote"}>{quote}</p>
                </div>
                )
            }
            <Button onClick={increment}>
                Siguiente
            </Button>
        </>

    );
};

export default MultipleCustomHooks;
