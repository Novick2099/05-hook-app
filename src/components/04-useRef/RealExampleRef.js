import React, {useState} from 'react';
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";
import {Button} from "@material-ui/core";

//todo renderizar componente de manera condicional
// previene error de que el componenete ya fue desmontado y la tarea asincrona aun continua
// cancelar subscripciones si el componente no esta montado
const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h2>Real Example Hook</h2>
            <hr/>
            {show && <MultipleCustomHooks/>}
            <Button id={"showButton"} onClick={()=>{setShow(!show);}}>
                Show / Hide
            </Button>

        </div>
    );
};

export default RealExampleRef;
