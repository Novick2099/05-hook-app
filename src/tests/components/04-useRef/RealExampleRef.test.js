import RealExampleRef from "../../../components/04-useRef/RealExampleRef";
import {mount, shallow} from "enzyme";
import React from "react";
import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks";

describe('Test in Real ExampleRef',() => {

  it('Should match to snapshot',()=>{
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should show <MultipleCustomHooks/>  when click on button',()=>{

    const wrapper = mount(<RealExampleRef />);
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    const showButton = wrapper.find('#showButton').last();
    showButton.simulate('click');
    wrapper.update();
    //todo buscar un componente en los test
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

  });

});