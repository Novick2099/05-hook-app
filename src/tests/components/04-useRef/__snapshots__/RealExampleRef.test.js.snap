// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Real ExampleRef Should match to snapshot 1`] = `
<div>
  <h2>
    Real Example Hook
  </h2>
  <hr />
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    Show / Hide
  </WithStyles(ForwardRef(Button))>
</div>
`;
